.github/workflows/build-push-containers.yml

```diff
@@ -0,0 +1,331 @@
+name: Build Push Containers
+
+on:
+  push:
```

Comment:
Do we really want to do this on every push? This could be down to each throwaway commit pushed into a development branch.

Reply:
Perhaps it should be filtered to the branch `main`, after you confirm that the CI workflow works in this branch.

No need to automatically push junky development images to Docker Hub.

Reply:
These go to ghcr.io and are important for testing. We could pattern match the commit content and use an `if: matched` expression. This would show when container checks are skipped s.t. container correctness can still be verified.